src/server/api/mutations/updateQuestionResponses.js

```diff
@@ -0,0 +1,115 @@
+import { log } from "../../../lib";
+import { assignmentRequiredOrAdminRole } from "../errors";
+import { cacheableData } from "../../models";
+const ActionHandlers = require("../../../integrations/action-handlers");
+
+export const updateQuestionResponses = async (
+  questionResponses,
+  campaignContactId,
+  loaders,
+  user
+) => {
+  let contact = await loaders.campaignContact.load(campaignContactId);
+  const campaign = await loaders.campaign.load(contact.campaign_id);
+  await assignmentRequiredOrAdminRole(
+    user,
+    campaign.organization_id,
+    contact.assignment_id,
+    contact
+  );
+
+  await cacheableData.questionResponse
+    .save(campaignContactId, questionResponses)
+    .catch(err => {
+      log.error(
+        `Error saving updated QuestionResponse for campaignContactID ${campaignContactId} questionResponses ${JSON.stringify(
+          questionResponses
+        )} error ${err}`
+      );
+    });
+
+  contact = await loaders.campaignContact.load(campaignContactId);
+
+  // The rest is for ACTION_HANDLERS
+  const interactionSteps = await cacheableData.campaign.dbInteractionSteps(
```

Comment:
In the move from the schema.js version the high conditional here:
https://github.com/MoveOnOrg/Spoke/pull/1543/files#diff-e3e0c0fe13e813c7748359d6a931a6faL1128
to test for any action handlers is meant to short-circuit any additional db calls if there aren't any action handlers configured.  Then we don't need any of this info.

Also, you're removing a cache optimization from the schema.js version:
`        const interactionSteps =  campaign.interactionSteps ||   (await cacheableData.campaign.dbInteractionSteps(campaign.id));`
skips loading them if the campaign was from a redis cache (which includes the interactionSteps in the cache object)

Reply:
I'll revise.